api/v1beta1/azuremanagedcontrolplane_webhook_test.go

```diff
@@ -1108,6 +1118,77 @@ func TestValidatingWebhook(t *testing.T) {
 			},
 			expectErr: false,
 		},
+		{
+			name: "Testing valid Marketplace Extension",
+			amcp: AzureManagedControlPlane{
+				ObjectMeta: getAMCPMetaData(),
+				Spec: AzureManagedControlPlaneSpec{
+					AzureManagedControlPlaneClassSpec: AzureManagedControlPlaneClassSpec{
+						Version: "v1.17.8",
+						MarketplaceExtensions: []MarketplaceExtension{
+							{
+								Name:          "extension1",
+								ExtensionType: ptr.To("test-type"),
+								Plan: &MarketplacePlan{
+									Name:      "test-plan",
+									Product:   "test-product",
+									Publisher: "test-publisher",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectErr: false,
```

Comment:
Out of scope for this PR, but do you think it'd be better to change the type to `expectedErr` and match the error message?

Reply:
Yeah I think that would be ideal. There are a ton of test cases in this file but I still think it'll be worth doing